modules/cudev/include/opencv2/cudev/grid/reduce_to_vec.hpp

```diff
@@ -183,12 +183,14 @@ __host__ void gridReduceToColumn_(const SrcPtr& src, GpuMat_<ResType>& dst, cons
     CV_Assert( getRows(mask) == rows && getCols(mask) == cols );
 
     dst.create(1, rows);
+    cuda::createContinuous(rows, 1, dst.type(), dst);
```

Comment:
In that case you need to remove ` dst.create` and `dst.reshape` calls. They are redundant.